File: java-basics/102/DefaultEncoding.java
import java.nio.charset.Charset;

public class DefaultEncoding {
    public static void main(String[] args) {
        // Pobieranie domyślnego systemu kodowania
        Charset defaultCharset = Charset.defaultCharset();

        // Wyświetlenie domyślnego kodowania
        System.out.println("Domyślny system kodowania plików: " + defaultCharset.displayName());

        // Opcjonalnie: Wyświetlenie dodatkowych szczegółów
        System.out.println("Czy domyślne kodowanie obsługuje Unicode? " + defaultCharset.contains(Charset.forName("UTF-8")));
    }
}

File: csharpagain/systemandstuff/defaultencoding.cs
namespace Program123
{
    using System;
    using System.Text;

    class DefaultEncoding
    {
        static void Main(string[] args)
        {
            // Pobranie domyślnego kodowania
            Encoding defaultEncoding = Encoding.Default;

            // Wyświetlenie domyślnego kodowania
            Console.WriteLine("Domyślny system kodowania plików: " + defaultEncoding.WebName);

            // Sprawdzenie obsługi Unicode
            bool isUnicode = defaultEncoding.WebName.Contains("utf");
            Console.WriteLine("Czy domyślne kodowanie obsługuje Unicode? " + isUnicode);
        }
    }

}